API.md

```diff
@@ -1,22 +1,26 @@
 # API Reference
-Include and instantiate the TheThingsNetwork class.
+Include and instantiate the TheThingsNetwork class. The constructor sets the value of the spreading factor, the front-side Bus and the frequency plan. It also initialize the library with the Streams it should communicate with.
 
 ```c
 #include <TheThingsNetwork.h>
-TheThingsNetwork ttn
+TheThingsNetwork ttn(Stream& modemStream, Stream& debugStream, fp_ttn_t fp, int sf = 7, int fsb = 2);
 ```
+- `Stream& modemStream`: Stream for the LoRa modem (for The Things Node/Uno use Serial1 and data rate 57600).
+- `Stream& debugStream`: Stream for write debug logs to (for The Things Node/Uno use Serial and data rate 9600).
+- `int sf = 7`: spreading factor(defaults to 7).
+- `int fsb = 2`: front-side bus(defaults to 2).
 
-## Method: init
-Initialise the library with the streams it should communicate with.
+##Method: configureChannels
+Initialize the device with a frequency plan (EU868 or US915). It sets the frequencies, data rate range, duty cycles of every channels.
 
 ```c
-void init(Stream& modemStream, Stream& debugStream);
+void configureChannels(int sf, int fsb)
 ```
 
-- `Stream& modemStream`: Stream for the LoRa modem (for The Things Node/Uno use `Serial1` and data rate `57600`).
-- `Stream& debugStream`: Stream for write debug logs to (for The Things Node/Uno use `Serial` and data rate `9600`).
+- `int sf`: spreading factor(defaults to 7), it decides which data rate you set.
+- `int fsb`: front-side bus(defaults to 2), it decides which channel should be open.
 
-Set the data rates with [`Serial.begin()`](https://www.arduino.cc/en/Serial/Begin) before passing them to `init()`.
+You choose which frequency plan you want by adding it in the constructor's argument, same goes for sf and fsb.
```

Comment:
If I need to set that via the constructor and sf and fsb as well, what us is this function? Explain.